src/components/safe-apps/SafeAppList.tsx

```diff
@@ -28,6 +37,22 @@ const SafeAppList = () => {
 
   const debouncedSearchQuery = useDebounce(searchQuery, 500)
 
+  const wcLink: LinkProps['href'] = useMemo(() => {
+    const wc = allSafeApps.find((app) => app.tags.includes(SafeAppsTag.WALLET_CONNECT))
+
+    if (!wc) {
+      return '#'
+    }
+
+    return {
+      pathname: AppRoutes.apps,
+      query: {
+        safe: router.query.safe,
+        appUrl: wc.url,
+      },
+    }
+  }, [allSafeApps, router.query.safe])
```

Comment:
I would extract this into a hook `useWalletConnectApp` like I did for the Transaction Builder in NewTxModal.